jwst/resample/resample.py

```diff
@@ -353,45 +377,71 @@ def resample_many_to_one(self, input_models):
+                    empty_model.var_flat = np.empty((1, 1))
+                    input_list.append(empty_model)
+                    input_models.shelve(model, i, modify=False)
+            self.blend_output_metadata(output_model, input_list)
+            del input_list
 
         # Initialize the output with the wcs
         driz = gwcs_drizzle.GWCSDrizzle(output_model, pixfrac=self.pixfrac,
                                         kernel=self.kernel, fillval=self.fillval)
 
         log.info("Resampling science data")
-        for img in input_models:
-            iscale = self._get_intensity_scale(img)
-            log.debug(f'Using intensity scale iscale={iscale}')
-            img.meta.iscale = iscale
-
-            inwht = resample_utils.build_driz_weight(img,
-                                                     weight_type=self.weight_type,
-                                                     good_bits=self.good_bits)
-            # apply sky subtraction
-            blevel = img.meta.background.level
-            if not img.meta.background.subtracted and blevel is not None:
-                data = img.data - blevel
-            else:
-                data = img.data.copy()
+        with input_models:
+            for img in input_models:
+                iscale = self._get_intensity_scale(img)
+                log.debug(f'Using intensity scale iscale={iscale}')
+                img.meta.iscale = iscale
 
-            xmin, xmax, ymin, ymax = resample_utils._resample_range(
-                data.shape,
-                img.meta.wcs.bounding_box
-            )
+                inwht = resample_utils.build_driz_weight(img,
+                                                        weight_type=self.weight_type,
+                                                        good_bits=self.good_bits)
+                # apply sky subtraction
+                blevel = img.meta.background.level
+                if not img.meta.background.subtracted and blevel is not None:
+                    data = img.data - blevel
+                else:
+                    data = img.data.copy()
 
-            driz.add_image(
-                data,
-                img.meta.wcs,
-                iscale=iscale,
-                inwht=inwht,
-                xmin=xmin,
-                xmax=xmax,
-                ymin=ymin,
-                ymax=ymax
-            )
-            del data, inwht
+                xmin, xmax, ymin, ymax = resample_utils._resample_range(
+                    data.shape,
+                    img.meta.wcs.bounding_box
+                )
+
+                driz.add_image(
+                    data,
+                    img.meta.wcs,
+                    iscale=iscale,
+                    inwht=inwht,
+                    xmin=xmin,
+                    xmax=xmax,
+                    ymin=ymin,
+                    ymax=ymax
+                )
+                del data, inwht
+                input_models.shelve(img)
```

Comment:
Is the `img.meta.iscale = iscale` line which prevents adding `modify=False` here? If so a follow-up issue/ticket would be helpful to see if we can remove the `img.meta.iscale` assignment.

Reply:
sounds good.  Do you think this is worth opening a ticket for?  if not, feel free to resolve this

Reply:
Opened ticket to track this: https://jira.stsci.edu/browse/JP-3713